metricbeat/helper/http.go

```diff
@@ -47,10 +47,14 @@ type HTTP struct {
 func NewHTTP(base mb.BaseMetricSet) (*HTTP, error) {
 	config := struct {
 		TLS             *tlscommon.Config `config:"ssl"`
+		ConnectTimeout  time.Duration     `config:"connect_timeout"`
 		Timeout         time.Duration     `config:"timeout"`
 		Headers         map[string]string `config:"headers"`
 		BearerTokenFile string            `config:"bearer_token_file"`
-	}{}
+	}{
+		ConnectTimeout: 2 * time.Second,
```

Comment:
What if someone sets ConnectionTimeout > Timeout or Timeout > Period? Should we allow that or is it an invalid config?

Reply:
yes. There's also this: https://golang.org/pkg/net/http/#Transport.MaxConnsPerHost

Reply:
> What if someone sets ConnectionTimeout > Timeout or Timeout > Period? Should we allow that or is it an invalid config?

At the moment somehow we are allowing `Timeout > Period` when no timeout is set (what would be like an infinite timeout). We could add some sanity checks for that and print at least a warning.

ConnectionTimeout > Timeout is allowed, but in that case the connection timeout won't ever happen. Not sure if we should do anything in that case, as much to print some info message to inform the user that the connection timeout will be ignored.

> There's also this: https://golang.org/pkg/net/http/#Transport.MaxConnsPerHost

@roncohen good one. How do you think we can use it here? I think we could set it to a high but sane value (like 20, or 100), but I wonder if it would make sense to expose it in the config.